drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -343,6 +343,42 @@ def data_collection_features
     end.compact
   end
 
+  def occurrence_point_forms
+    project.occurrence_point_form_instances.map do |instance|
+      matches = instance.project_and_enrollment_match(project: project, enrollment: self).present?
+      has_any_data = false
+
+      unless matches # Only do this computation if the instance doesn't match this enrollment
```

Comment:
To reduce indentation, we could bump up the `next`:

```ruby
project.occurrence_point_form_instances.map do |instance|
  matches = instance.project_and_enrollment_match(project: project, enrollment: self).present?

  # This instance is enabled for this project _and_ matches this Enrollment (for "Data Collected About"), so we know it should be shown for this Enrollment. Proceed
  next instance if matches

  # This instance is enabled for this project BUT doesn't match this Enrollment. Determine if it has any data, in which case we will show it anyway.
  ... has_any_data computation logic

```

I think this logic is a little different from what we discussed, though. My understanding of where we landed was that the "occurrence point form" should _always_ be shown if it has any data (specifically for HUD elements -- MID/DOE/PATH status), even if that form is not enabled in the given project.

Reply:
Ah I see, yep that tracks. Thanks. I'll take another look at it today and create a test where there is legacy data in a non-enabled form.